WalletWasabi/TorSocks5/TorSocks5Client.cs

```diff
@@ -205,61 +205,77 @@ internal async Task ConnectToDestinationAsync(EndPoint destination, bool isRecur
 
-			var dstAddr = new AddrField(host);
-			DestinationHost = dstAddr.DomainOrIPv4;
+					var dstPort = new PortField(port);
+					DestinationPort = dstPort.DstPort;
 
-			var dstPort = new PortField(port);
-			DestinationPort = dstPort.DstPort;
+					var connectionRequest = new TorSocks5Request(cmd: CmdField.Connect, dstAddr, dstPort);
+					var sendBuffer = connectionRequest.ToBytes();
 
-			var connectionRequest = new TorSocks5Request(cmd, dstAddr, dstPort);
-			var sendBuffer = connectionRequest.ToBytes();
+					var receiveBuffer = await SendAsync(sendBuffer, isRecursiveCall: isRecursiveCall).ConfigureAwait(false);
 
-			var receiveBuffer = await SendAsync(sendBuffer, isRecursiveCall: isRecursiveCall).ConfigureAwait(false);
+					var connectionResponse = new TorSocks5Response();
+					connectionResponse.FromBytes(receiveBuffer);
 
-			var connectionResponse = new TorSocks5Response();
-			connectionResponse.FromBytes(receiveBuffer);
+					if (connectionResponse.Rep != RepField.Succeeded)
+					{
+						// https://www.ietf.org/rfc/rfc1928.txt
+						// When a reply(REP value other than X'00') indicates a failure, the
+						// SOCKS server MUST terminate the TCP connection shortly after sending
+						// the reply.This must be no more than 10 seconds after detecting the
+						// condition that caused a failure.
+						DisposeTcpClient();
+						Logger.LogWarning($"Connection response indicates a failure. Actual response is: '{connectionResponse.Rep}'.");
+						throw new TorSocks5FailureResponseException(connectionResponse.Rep);
+					}
 
-			if (connectionResponse.Rep != RepField.Succeeded)
+					// Do not check the Bnd. Address and Bnd. Port. because Tor does not seem to return any, ever. It returns zeros instead.
+					// Generally also do not check anything but the success response, according to Socks5 RFC
+
+					// If the reply code(REP value of X'00') indicates a success, and the
+					// request was either a BIND or a CONNECT, the client may now start
+					// passing data. If the selected authentication method supports
+					// encapsulation for the purposes of integrity, authentication and / or
+					// confidentiality, the data are encapsulated using the method-dependent
+					// encapsulation.Similarly, when data arrives at the SOCKS server for
+					// the client, the server MUST encapsulate the data as appropriate for
+					// the authentication method in use.
+				}
+			}
+			catch (Exception e)
 			{
-				// https://www.ietf.org/rfc/rfc1928.txt
-				// When a reply(REP value other than X'00') indicates a failure, the
-				// SOCKS server MUST terminate the TCP connection shortly after sending
-				// the reply.This must be no more than 10 seconds after detecting the
-				// condition that caused a failure.
-				DisposeTcpClient();
-				throw new TorSocks5FailureResponseException(connectionResponse.Rep);
+				Logger.LogError("Exception was thrown when connecting to destination.", e);
+			}
```

Comment:
This is a breaking change and it is not clear how it affects the rest of the code. rethrow the exception or explain how swallowing it does not affect the current behavior.

Reply:
Fixed. Added re-throw.